JFRWebSocket.m

```diff
@@ -195,10 +195,11 @@ - (void)createHTTPRequest
                                      (__bridge CFStringRef)headerWSHostName,
                                      (__bridge CFStringRef)[NSString stringWithFormat:@"%@:%@",self.url.host,port]);
     
-    for(NSString *key in self.headers) {
+    NSDictionary *heads = [self.headers copy];
```

Comment:
Since this method was moved unto the client thread, the `copy` is a bit pedantic.

I had suggested adding `copy` to the `addHeader:` method itself, which would make `self.headers` immutable and totally safe under any situation.